### ISSUE TITLE
"Delete all unused Declarations" removed necessary parameters from callbacks

### ISSUE THREAD

**Issue by bcrobinson** [Microsoft/TypeScript#25401]
**TypeScript Version:**  2.9.2 (also confirmed 3.0.0-dev.20180703)
Issue Type : **Bug**

**Search Terms*:*
Refactor
Delete all unused Declarations
fixUnusedIdentifier

**Code**
When running the remove unused declarations refactor in VSCode it will remove unused parameters from any callbacks even when they are not the last position, which causes code to behave differently.

**Initial code**
```ts
import * as Q from 'q';

let items = ["a", "b", "c"];

items.map((c, i) => i);
```
**Actual Code after refactor**
After running "Delete all unused Declarations" refactor on the unused import line ends up with:
```ts
let items = ["a", "b", "c"];

items.map((i) => i);
```

The refactor removed he "c" parameter, which changed the behaviour of the code.

**Expected behaviour:**
Expected the unused "c" parameter to be left alone (or replaced in "_", just not removed).
```ts
let items = ["a", "b", "c"];

items.map((c, i) => i);
```

**Related Issues:**
Something similar was mentioned in https://github.com/Microsoft/TypeScript/issues/24789#issuecomment-395690600 @andy-ms seems to have done a change in https://github.com/Microsoft/TypeScript/pull/25011 related to that comment, but from what I can tell that seems to be for also removing the unused parameters from the function body instead.

**Comment by andy-ms** [comment]
Doesn't reproduce for me with `typescript@3.0.0-dev.20180703`. Are you sure your editor is [set up](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions) to use the locally installed typescript?

**Comment by bcrobinson** [comment]
Yes it seems that VSCode was still using v2.9.2.
I've tested this in a brand new project and confirmed that it's fixed in the latest daily build (`typescript@3.0.0-dev.20180704`).
